planner/core/logical_plan_builder.go

```diff
@@ -2822,7 +2822,7 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 				if err != nil {
 					return nil, err
 				}
-				colExpr.VirtualExpr = expr
+				colExpr.VirtualExpr = expr.Clone()
```

Comment:
Check `expr != nil` before use?

Reply:
If the err is nil, the `expr` shouldn't be nil. This assumption is also followed by other usages.